pkg/collections.go

```diff
@@ -0,0 +1,264 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Gergely Brautigam
+//
+package pkg
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/arangodb/go-driver"
+	"github.com/briandowns/spinner"
+	"github.com/cenkalti/backoff"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+)
+
+// copyCollections copies all collections for a database.
+func (c *copier) copyCollections(ctx context.Context, db driver.Database) error {
+	log := c.Logger
+	var (
+		sourceDB      driver.Database
+		list          []driver.Collection
+		destinationDB driver.Database
+	)
+	c.backoffCall(ctx, func() error {
```

Comment:
Don't like the grouping of 3 functions (to 2 different arango addresses) in a single backoff.
If destinationClient.Database fails, there is no need to retry sourceClient.Database